c11threads.h

```diff
@@ -1,27 +1,67 @@
 /*
-Author: John Tsiombikas <nuclear@member.fsf.org>
+c11threads
+
+Authors:
+John Tsiombikas <nuclear@member.fsf.org>
+Oliver Old <oliver.old@outlook.com>
 
 I place this piece of code in the public domain. Feel free to use as you see
-fit.  I'd appreciate it if you keep my name at the top of the code somehwere,
-but whatever.
+fit. I'd appreciate it if you keep my name at the top of the code somewhere, but
+whatever.
 
 Main project site: https://github.com/jtsiomb/c11threads
 */
 
 #ifndef C11THREADS_H_
 #define C11THREADS_H_
 
+#if !defined(_WIN32) || defined(C11THREADS_PTHREAD_WIN32)
 #include <stdint.h>
-#include <time.h>
 #include <errno.h>
 #include <pthread.h>
 #include <sched.h>	/* for sched_yield */
 #include <sys/time.h>
+#endif
+#include <time.h>
 
-#define thread_local	_Thread_local
-#define ONCE_FLAG_INIT	PTHREAD_ONCE_INIT
+#ifndef thread_local
```

Comment:
Is it really necessary to check the stdc version and/or C++ version? Does it cause a problem if we unconditionally define thread_local?

Reply:
Defining the macro for C23 or newer would be non-standard, so I don't think that would be a good idea. Likewise, I believe defining a macro whose name is identical to a reserved keyword is not great.

Reply:
I'm not fussed about that. We're not writing a specific C-standard libc after all. And we are defining all those things even if the system libc is C99 and would not have them anyway.

Reply:
I'll define it always then unless it was already defined. Users can `#undef` it if they don't need it.